compiler/kernel/op.go

```diff
@@ -713,11 +717,24 @@ func (b *Builder) compilePoolScan(scan *dag.PoolScan) (zbuf.Puller, error) {
 	return meta.NewSequenceScanner(b.rctx, slicer, pool, nil, nil, b.progress), nil
 }
 
-func (b *Builder) PushdownOf(e dag.Expr) *Filter {
-	if e == nil {
+func (b *Builder) NewPushdown(e dag.Expr, projection []field.Path) *pushdown {
```

Comment:
But: Return a zbuf.Pushdown here since it's a little opaque for a public method to return an unexported type.

Reply:
👍